### ISSUE TITLE
ResizeObserver is not defined

### ISSUE THREAD

**Issue by aspnetde** [uiwjs/react-markdown-editor#226]
When adding the markdown editor to a current NextJs client component, building fails with a rather generic `ReferenceError: ResizeObserver is not defined`.

I don't have time for a reproduction test right now (sorry), but a quick workaround is using the polyfill as described here: https://stackoverflow.com/a/78560100/271150

**Comment by jaywcjlove** [comment]
@aspnetde  You might need to move it to client-side rendering instead of server-side. Add `'use client';` at the top of the code.

**Comment by joelrb** [comment]
> @aspnetde You might need to move it to client-side rendering instead of server-side. Add `'use client';` at the top of the code.

I have this issue. Both the parent and the component have "use client"; at the top of code. First time to use Markdown.

` <MarkdownEditor
        value={description}
        enablePreview
        enableScroll
        height="600px"
        className="w-full"
        onChange={(value, viewUpdate) => setDescription(value)}
  />`

![Screenshot 2024-09-10 164418](https://github.com/user-attachments/assets/aa34ee5c-8351-4fd3-bdf1-8c81b6315159)

**Comment by joelrb** [comment]
Indeed, adding this to the component seems to have fixed the issue.

import ResizeObserver from 'resize-observer-polyfill';
global.ResizeObserver = ResizeObserver; 


> When adding the markdown editor to a current NextJs client component, building fails with a rather generic `ReferenceError: ResizeObserver is not defined`.
> 
> I don't have time for a reproduction test right now (sorry), but a quick workaround is using the polyfill as described here: https://stackoverflow.com/a/78560100/271150
